compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala

```diff
@@ -203,10 +211,49 @@ case class ApplicationCompiler(typeAliases: Map[String, Type],
         }
       case _ => Map.empty
     }
-    val bundledDepends = runtimeAsset match {
-      case Some(jsv) => Map("bundledDepends" -> JsArray(Vector(jsv)))
-      case None      => Map.empty
+
+    // If dependency on Docker image file in another project, clone
+    applet.container match {
+      case DxFileDockerImage(_, dxfile) => {
+        val id = dxfile.id
+
+        // If source project not specified for Docker image file, try to find it
+        val sourceProject = dxfile.project.getOrElse(
+            Try {
+              dxApi.getObject(id).describe(Set(Field.Project)) match {
+                case f: DxFileDescribe => DxProject.apply(f.project)(dxApi)
```

Comment:
Use `dxApi.project(f.project)` - generally, the API methods should be used rather than creating objects directly.